File: e2e/selenide-junit5-selenium-grid/src/test/java/interaction/page/InfoTaskModal.java
package interaction.page;

import com.codeborne.selenide.SelenideElement;
import data.TaskPriority;
import data.TaskStatus;

import static com.codeborne.selenide.Selenide.$;

public class InfoTaskModal {
    public SelenideElement title = $("[data-testid='modal-title']");
    public SelenideElement validIcon = $("[data-testid='valid']");
    public SelenideElement descriptionField = $("[data-testid='description']");

    public SelenideElement statusTag(TaskStatus status) {
        return $("[data-testid='status-tag-" + status + "']");
    }

    public SelenideElement priorityTag(TaskPriority priority) {
        return $("[data-testid='priority-tag-" + priority + "']");
    }
}

File: demo-android/app/src/androidTest/java/com/example/demo/e2e/interaction/page/InfoTaskModal.kt
package com.example.demo.e2e.interaction.page

import androidx.compose.ui.test.junit4.AndroidComposeTestRule
import com.example.demo.MainActivity
import com.example.demo.data.model.TaskPriority
import com.example.demo.data.model.TaskStatus

class InfoTaskModal(
    rule: AndroidComposeTestRule<*, MainActivity>,
) : ComposePage(rule) {
    fun title() = node("modal-title")

    fun validIcon() = node("valid")

    fun descriptionField() = node("description")

    fun statusTag(status: TaskStatus) = node("status-tag-${status.name}")

    fun priorityTag(priority: TaskPriority) = node("priority-tag-${priority.name}")

    fun waitUntilVisible(timeoutMillis: Long = 15_000) {
        waitUntilPresent("modal-title", timeoutMillis)
        waitUntilAbsent("loading-spinner", timeoutMillis)
    }
}
